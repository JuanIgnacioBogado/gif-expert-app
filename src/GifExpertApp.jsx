import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = category => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onAddCategory={onAddCategory} />

      {categories?.map(category => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
