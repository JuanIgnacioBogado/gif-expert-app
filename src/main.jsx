import { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';

import { GifExpertApp } from './GifExpertApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GifExpertApp />
  </StrictMode>
);
