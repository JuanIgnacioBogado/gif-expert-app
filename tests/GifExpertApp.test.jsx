import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('GifExpertApp', () => {
  test('should to match snapshot', () => {
    const { container } = render(<GifExpertApp />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText(/GifExpertApp/i));
  });
});
