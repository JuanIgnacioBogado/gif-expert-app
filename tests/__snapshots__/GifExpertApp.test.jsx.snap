// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should to match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      name="categoryInput"
      placeholder="Buscar gifs"
      type="text"
    />
  </form>
  <h2>
    Cargando...
  </h2>
</div>
`;
